netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -46,121 +46,142 @@
 import io.netty.handler.codec.http2.Http2Headers;
 import io.netty.handler.codec.http2.Http2Stream;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import javax.net.ssl.SSLSession;
 
 /**
  * Server stream for a Netty HTTP2 transport.
  */
-class NettyServerStream extends AbstractServerStream<Integer> {
+class NettyServerStream extends AbstractServerStream {
+  private static final Logger log = Logger.getLogger(NettyServerStream.class.getName());
 
+  private final Sink sink = new Sink();
+  private final TransportState state;
   private final Channel channel;
-  private final NettyServerHandler handler;
-  private final Http2Stream http2Stream;
   private final WriteQueue writeQueue;
   private final Attributes attributes;
 
-  NettyServerStream(Channel channel, Http2Stream http2Stream, NettyServerHandler handler,
-                    int maxMessageSize) {
-    super(new NettyWritableBufferAllocator(channel.alloc()), maxMessageSize);
-    this.writeQueue = handler.getWriteQueue();
+  public NettyServerStream(Channel channel, TransportState state) {
+    super(new NettyWritableBufferAllocator(channel.alloc()));
+    this.state = checkNotNull(state, "transportState");
```

Comment:
I purposefully was more verbose in the string, since "state must not be null" can be misunderstood as being useless (which state‽).